File: IsSubsequence_392.java
class Solution {
    public static boolean isSubsequence(String s, String t) {
        char[] charS = s.toCharArray();
        char[] charT = t.toCharArray();
        
        if(charS.length == 0) return false;
        int i = 0;
        int j = 0;
        while(j < charT.length) {
            if(charS[i] == charT[j]) {
                i++;
                j++;
                if(i >= charS.length) return true;
            } else {
                j++;
            } 
        }
        return false;   
       
    }
    
    public static void main(String[] args) {
        System.out.println(isSubsequence("", "ahbgdc"));
    }
}

File: IsSubsequence_392.py
class Solution:
    def isSubsequence(self, s, t):
        n = len(s)
        i = 0
        for ch in t:
            if i < n and s[i] == ch:
                i += 1
        return i == n


if __name__ == '__main__':
    solu = Solution()
    s = "abc"
    t = "ahbgdc"
    print(solu.isSubsequence(s, t))